spec.html

```diff
@@ -20653,20 +21261,269 @@ <h1>Static Semantics: PrototypePropertyNameList</h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-class-definitions-static-semantics-allprivateidentifiersvalid" type="sdo" aoid="AllPrivateIdentifiersValid">
+      <h1>Static Semantics: AllPrivateIdentifiersValid</h1>
+      <p>With parameter _names_.</p>
+
+      <p>Every grammar production alternative in this specification which is not listed below implicitly has the following default definition of AllPrivateIdentifiersValid:</p>
+      <emu-alg>
+        1. For each child node _child_ of this Parse Node, do
+          1. If _child_ is an instance of a nonterminal, then
+            1. If AllPrivateIdentifiersValid of _child_ with argument _names_ is *false*, return *false*.
+        1. Return *true*.
+      </emu-alg>
+
+      <emu-grammar>MemberExpression : MemberExpression `.` PrivateIdentifier</emu-grammar>
+      <emu-alg>
+        1. If _names_ contains the StringValue of |PrivateIdentifier|, then
+          1. Return AllPrivateIdentifiersValid of |MemberExpression| with argument _names_.
+        1. Return *false*.
+      </emu-alg>
+
+      <emu-grammar>CallExpression : CallExpression `.` PrivateIdentifier</emu-grammar>
+      <emu-alg>
+        1. If _names_ contains the StringValue of |PrivateIdentifier|, then
+          1. Return AllPrivateIdentifiersValid of |CallExpression| with argument _names_.
+        1. Return *false*.
+      </emu-alg>
+
+      <emu-grammar>OptionalChain : `?.` PrivateIdentifier</emu-grammar>
+      <emu-alg>
+        1. If _names_ contains the StringValue of |PrivateIdentifier|, return *true*.
+        1. Return *false*.
+      </emu-alg>
+
+      <emu-grammar>OptionalChain : OptionalChain `.` PrivateIdentifier</emu-grammar>
+      <emu-alg>
+        1. If _names_ contains the StringValue of |PrivateIdentifier|, then
+          1. Return AllPrivateIdentifiersValid of |OptionalChain| with argument _names_.
+        1. Return *false*.
+      </emu-alg>
+
+      <emu-grammar>ClassBody : ClassElementList</emu-grammar>
+      <emu-alg>
+        1. Let _newNames_ be a copy of _names_.
+        1. Append to _newNames_ the elements of PrivateBoundIdentifiers of |ClassBody|.
```

Comment:
```suggestion
        1. Let _newNames_ be a copy of _names_ with the elements of PrivateBoundIdentifiers of |ClassBody| appended.
```
this shorter form is used already in a few places in the spec.

Reply:
Eh, I think the alias improves readability in this case.